stor/obs.py

```diff
@@ -450,6 +452,22 @@ def close(self):
                 self.flush()
             self._buffer.close()
         self.closed = True
+        # we want to wait_on_close on DXPath only if no error was thrown while writing the file
+        if not any(sys.exc_info()):  # pragma: no cover
+            self.dx_close_helper()
+
+    def dx_close_helper(self):
+        if isinstance(self._path, stor.dx.DXPath):
+            wait_on_close = stor.settings.get()['dx']['wait_on_close']
+            if wait_on_close:
+                f = dxpy.DXFile(dxid=self._path.canonical_resource,
```

Comment:
btw - wouldn't it be more performant to just say `self._path.canonical_path.resource` and `self._path.canonical_path.project`

Reply:
It is the same performance wise. canonical_path calls canonical_resource and canonical_project each once, and caches them.

Reply:
gotcha